tutorials/src/bin/basic-tutorial-2.rs

```diff
@@ -0,0 +1,59 @@
+extern crate gstreamer as gst;
+
+use gst::GstObjectExt;
+use gst::ElementExt;
+use gst::MessageView;
+
+fn main() {
+    // Initialize GStreamer
+    gst::init().unwrap();
+
+    if let (Some(source), Some(sink)) = (
+    // Create the elements
+    gst::ElementFactory::make("videotestsrc", "source"),
+    gst::ElementFactory::make("autovideosink", "sink")) {
+        // Create the empty pipeline
+        let pipeline = gst::Pipeline::new("test-pipeline");
+
+        // Build the pipeline
+        gst::BinExtManual::add_many(
+            &pipeline,
+            &[&source, &sink]).unwrap();
+        if let Err(_) = gst::ElementExt::link(&source, &sink) {
+            eprintln!("Elements could not be linked.");
+            ::std::process::exit(-1);
+        }
+
+        // Modify the source's properties
+        gst::GObjectExtManualGst::set_property_from_str(
+            &source,
+            "pattern",
+            "0");
```

Comment:
You can even give the pattern a name here: 0 is "smpte"

Reply:
This part is mimicking: g_object_set (source, "pattern", 0, NULL);

My understanding was that "pattern" is the property to set, and "0" is the value for that property.

Could you explain what you mean by "give the pattern a name"?

Reply:
The equivalent to the C code would be with `set_property` but we can do better here. Either you could use that in combination with `glib::EnumClass` (see playbin.rs for an example with flags, enums work very similar). That would be most explicit. Or you could use `source.set_property_from_str("pattern", "smpte") `

The advantage with using the string is that it is more descriptive. You don't have to look up in the docs that 0 is the SMPTE pattern but the code just says so. And you could replace it directly with e.g. " snow" without looking up the numerical value of it.